src/libsyntax/parse/lexer/unicode_chars.rs

```diff
@@ -0,0 +1,156 @@
+                                                      (14, "Fraction Slash"),
+                                                      (14, "Box Drawings Light Diagonal Upper Right To Lower Left"),
+                                                      (14, "Mathematical Rising Diagonal"),
+                                                      (14, "Big Solidus"), (14, "Cjk Stroke Sp"),
+                                                      (14, "Vertical Kana Repeat Mark Upper Half"),
+                                                      (14, "Cjk Unified Ideograph-4E3F"),
+                                                      (14, "Kangxi Radical Slash"),
+                                                      (15, "Fullwidth Reverse Solidus"),
+                                                      (15, "Small Reverse Solidus"),
+                                                      (15, "Set Minus"),
+                                                      (15, "Mathematical Falling Diagonal"),
+                                                      (15, "Reverse Solidus Operator"),
+                                                      (15, "Big Reverse Solidus"),
+                                                      (15, "Cjk Stroke D"),
+                                                      (15, "Cjk Unified Ideograph-4E36"),
+                                                      (15, "Kangxi Radical Dot"),
+                                                      (16, "Latin Small Letter Um"),
+                                                      (17, "Hebrew Letter Alternative Plus Sign"),
+                                                      (18, "Single Left-Pointing Angle Quotation Mark"),
+                                                      (18, "Heavy Left-Pointing Angle Quotation Mark Ornament"),
+                                                      (18, "Modifier Letter Left Arrowhead"),
+                                                      (19, "Lisu Punctuation Full Stop"),
+                                                      (20, "Single Right-Pointing Angle Quotation Mark"),
+                                                      (20, "Heavy Right-Pointing Angle Quotation Mark Ornament"),
+                                                      (20, "Modifier Letter Right Arrowhead"),
+                                                      (1, "Coptic Capital Letter Dialect-P Ni"),
+                                                      (6, "Latin Capital Letter Glottal Stop"),
+                                                      (14, "Coptic Capital Letter Old Coptic Esh"), ];
+
+pub fn check_for_substitution(ch: char) -> Option<String> {
+    UNICODE_ARRAY
+    .iter()
+    .position(|&c| c == ch)
+    .map(|idx| {
+        let (ascii_pos, u_name) = SUB_ARRAY[idx];
+        let (sub_char, sub_name) = ASCII_ARRAY[ascii_pos];
+        format!("Unicode character '{}' ({}) looks much like a '{}' ({}), but it's not.",
```

Comment:
I don't think the "a" is necessary. "Unicode char 'f' (...) looks much like 'g' (..), but it's not"